### PR TITLE
Blob Generator

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
Demo is here: https://1hqidr.csb.app/

### CODE REVIEW

**Comment by TheSonOfThomp** [comment]
Status update: Generates shapes constructed from only small circles (i.e. no 2x2 circles yet)

Screen Shot 2022-07-19 at 4 34 20 PM

**Comment by TheSonOfThomp** [comment]
Update: Generates large (2x2) circles now

Screen Shot 2022-07-20 at 6 56 21 PM

**Comment by TheSonOfThomp** [comment]
Interactive generator story

Screen Shot 2022-07-22 at 4 51 50 PM

File: packages/brand-shape/src/Blob.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  return (
+    <svg
+      viewBox="0 0 8 8"
+      width="100%"
+      height="100%"
+      className={className}
+      {...rest}
+    >
+      {/* DEBUG */}
```
Comment (brookescarlett) [inline packages/brand-shape/src/Blob.tsx:27]: 😂

**Comment by brookescarlett** [comment]
Keep the branch but maybe we close the PR? Unless we're actually looking for feeedback / to merge this in

File: packages/brand-shape/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+```
+
+## Example
+
+**Output HTML**
+
+## Properties
+
+| Prop | Type | Description | Default |
+| ---- | ---- | ----------- | ------- |
```
Comment (spark33) [inline packages/brand-shape/README.md:28]: Need prop table, and some documentation on how the `o` vs. `O` in the `shape` prop should be expected to behave. 

There are also a few shapes that feel like they should work, but maybe they're not possible due to technical constraints - like:
```
<Blob
shape={[
["o","o","o","o"],
["o"," "," ","o"],
["o"," "," ","o"],
["o","o","o","o"]
]}
/>
```

File: packages/brand-shape/src/Blob.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          ['o', ' ', ' ', 'o'],
+        ]}
+      />,
+    );
+
+    const blob = getByTestId(container, 'blob');
+    expect(blob).toBeInTheDocument();
+  });
+
+  // TODO: add more tests
```
Comment (spark33) [inline packages/brand-shape/src/Blob.spec.tsx:24]: probably would want to lean on Chromatic more than functional tests - just wouldn't make sense for a package like this

File: packages/brand-shape/src/Blob.tsx (hunk 1 above)
Comment (spark33) [reply packages/brand-shape/src/Blob.tsx:29]: If I were to review this component for production purposes, I would propose that we include this logic in a `BlobDebugger` wrapper component that functions similarly to the V11Table's V11Adapter component where these svg elements can be injected into the regular blob component.

File: packages/brand-shape/src/generateBlobPath.ts [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+          y,
+          b1x: x,
+          b1y: y - bezY,
+          b2x: xNext === x ? x : xNext > x ? xNext - bezX : xNext + bezX, // next v is ? right : left: ;
+          b2y: yNext,
+        };
+      }
+    }
+  });
+}
```
Comment (spark33) [inline packages/brand-shape/src/generateBlobPath.ts:290]: These functions should be tested

File: packages/brand-shape/src/types.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+export const PrevDirection: Record<Direction, Direction> = {
+  top: 'topLeft',
+  topRight: 'top',
+  right: 'topRight',
+  bottomRight: 'right',
+  bottom: 'bottomRight',
+  bottomLeft: 'bottom',
+  left: 'bottomLeft',
+  topLeft: 'left',
+} as const;
```
Comment (spark33) [inline packages/brand-shape/src/types.ts:134]: These could probably just be calculated from an array and indexed by `(x,y)` coords. e.g. topLeft would be `-1, -1`.

File: packages/brand-shape/src/utils.ts [hunk 6]
```diff
@@ -0,0 +1,368 @@
...
+    };
+  }
+
+  return {
+    top: [x, y - 1],
+    right: [x + 1, y],
+    bottom: [x, y + 1],
+    left: [x - 1, y],
+  };
+};
```
Comment (spark33) [inline packages/brand-shape/src/utils.ts:316]: Using `(x,y)` values for the directions would probably help make these functions shorter too.

**Review by spark33 — CHANGES_REQUESTED** [review]
There are few minor comments on code conciseness, but those aren't important.

What probably matters more is:
1) getting some documentation and expected usage guidelines on the README, 
2) testing the functional side of the utilities and `shape` prop, 
3) fixing bugs with the SVG generation,
4) moving the `debug` functionality to an internal component wrapper,
5) and getting clearance from any branding team that would be interested in this.

I think aside from these considerations, the code mostly looks good to me and pretty close to being released. I'm thinking I could put this on the Figma plugin to help designers generate blob shapes quickly.

**Pull request closed without merging**